projects/kit/components/input-slider/input-slider.component.ts

```diff
@@ -0,0 +1,145 @@
+import type {WritableSignal} from '@angular/core';
+import {
+    computed,
+    ContentChild,
+    Directive,
+    effect,
+    inject,
+    INJECTOR,
+    signal,
+} from '@angular/core';
+import {toObservable, toSignal} from '@angular/core/rxjs-interop';
+import type {TuiValueTransformer} from '@taiga-ui/cdk/classes';
+import {TuiNonNullableValueTransformer} from '@taiga-ui/cdk/classes';
+import {TUI_ALLOW_SIGNAL_WRITES} from '@taiga-ui/cdk/constants';
+import {tuiIsElement, tuiIsInput} from '@taiga-ui/cdk/utils/dom';
+import {tuiClamp} from '@taiga-ui/cdk/utils/math';
+import {tuiIsPresent} from '@taiga-ui/cdk/utils/miscellaneous';
+import {TuiTextfieldComponent} from '@taiga-ui/core/components/textfield';
+import {
+    TuiInputNumber,
+    tuiInputNumberOptionsProvider,
+} from '@taiga-ui/kit/components/input-number';
+import {
+    TuiSliderComponent,
+    TuiSliderKeyStepsBase,
+    tuiSliderOptionsProvider,
+} from '@taiga-ui/kit/components/slider';
+import {filter, map, pairwise, startWith, switchMap} from 'rxjs';
+
+@Directive({
+    standalone: true,
+    selector: 'tui-textfield[tuiInputSlider]',
+    providers: [
+        tuiSliderOptionsProvider({trackColor: 'transparent'}),
+        tuiInputNumberOptionsProvider({
+            valueTransformer: new TuiNonNullableValueTransformer(),
+        }),
+    ],
+    host: {
+        '(input)': 'onSliderInput($event)',
+        '(change)': 'onSliderInput($event)',
+        '(click)': 'textfield.input.nativeElement.focus()',
+        '(focusout)': 'onBlur()',
+        '(keydown.arrowUp)': 'onStep(1)',
+        '(keydown.arrowDown)': 'onStep(-1)',
+    },
+})
+export class TuiInputSliderDirective {
+    private readonly injector = inject(INJECTOR);
+    private keyStepsTransformer: WritableSignal<TuiValueTransformer<
+        number,
+        number
+    > | null> = signal(null);
+
+    private readonly inputNumber = signal<TuiInputNumber | null>(null);
+    private readonly slider = signal<TuiSliderComponent | null>(null);
+    protected readonly textfield = inject(TuiTextfieldComponent);
+
+    protected min = computed(() => this.inputNumber()?.min() ?? 0);
+    protected max = computed(() => this.inputNumber()?.max() ?? 100);
+    protected step = computed((slider = this.slider()) =>
+        slider && this.keyStepsTransformer() // For non-linear slider step means percentage
+            ? ((this.max() - this.min()) / 100) * slider.step()
+            : (slider?.step() ?? 1),
+    );
+
+    protected disabled = computed(
```

Comment:
Can we make it `interactive` and reversed, since it's about both disabled and readOnly?